starknet_py/net/models/transaction.py

```diff
@@ -1,28 +1,244 @@
+        """
+
+    @abstractmethod
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network - a unique identifier of the
+        transaction. See :ref:`compute_transaction_hash` docstring for more details.
+        """
+
+
+@dataclass(frozen=True)
+class AccountTransaction(Transaction):
+    """
+    Represents a transaction in the StarkNet network that is originated from an action of an
+    account.
+    """
+
+    max_fee: int = field(metadata={"marshmallow_field": Felt()})
+    signature: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    nonce: int = field(metadata={"marshmallow_field": Felt()})
+
+
+@dataclass(frozen=True)
+class Declare(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a declaration of a StarkNet contract
+    class.
+    """
+
+    contract_class: ContractClass
+    # The address of the account contract sending the declaration transaction.
+    sender_address: int = field(metadata={"marshmallow_field": Felt()})
+
+    @property
+    def tx_type(self) -> TransactionType:
+        return TransactionType.DECLARE
+
+    @marshmallow.post_dump
+    def compress_program_post_dump(
+        self, data: Dict[str, Any], many: bool, **kwargs
+    ) -> Dict[str, Any]:
+        data["type"] = "DECLARE"
+        return compress_program_post_dump(data=data, many=many)
+
+    @marshmallow.pre_load
+    def decompress_program(
+        self, data: Dict[str, Any], many: bool, **kwargs
+    ) -> Dict[str, Any]:
+        return decompress_program(data=data, many=many)
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network.
+        """
+        return compute_declare_transaction_hash(
+            contract_class=self.contract_class,
+            chain_id=chain_id.value,
+            sender_address=self.sender_address,
+            max_fee=self.max_fee,
+            version=self.version,
+            nonce=self.nonce,
+        )
+
+
+@dataclass(frozen=True)
+class DeployAccount(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a deployment of a StarkNet account
+    contract.
+    """
+
+    class_hash: int = field(metadata={"marshmallow_field": Felt()})
+    contract_address_salt: int = field(metadata={"marshmallow_field": Felt()})
+    constructor_calldata: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    version: int = field(metadata={"marshmallow_field": Felt()})
+
+    @property
+    def tx_type(self) -> TransactionType:
+        return TransactionType.DEPLOY_ACCOUNT
+
+    @marshmallow.post_dump
+    def post_dump(self, data: Dict[str, Any], many: bool, **kwargs) -> Dict[str, Any]:
+        data["type"] = "DEPLOY_ACCOUNT"
+        return data
```

Comment:
What's the purpose of this?

Reply:
the problem is that marshmallow does not serialize the "tx_type" field and we have to do it manually. There could be another solution, but I couldn't find it. I'm open to suggestions

Reply:
`tx_type` is a method marked as a `property` and this is probably the problem

Reply:
Ok, so now there is a field `type` not `tx_type` and it is automatically serialized by marshmallow